### PR TITLE
Finish PR #2432: Imagenet example updates + basic testing

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Finishes an old PR #2432 by @ruotianluo

- standardize with other examples
- a test that runs for 1 epoch making sure the script can run
- simplifications

### CODE REVIEW

**Comment by Borda** [comment]
You can still use this one... =)

File: pl_examples/test_examples.py [hunk 1]
```diff
@@ -25,6 +28,35 @@ def test_gpu_template(cli_args):
...
+
+    cli_args = cli_args.split(' ') if cli_args else []
+    cli_args += ['--data-path', str(tmpdir)]
+    cli_args += ['--default_root_dir', str(tmpdir)]
+
+    with mock.patch("argparse._sys.argv", ["any.py"] + cli_args):
+        run_cli()
+
+
 # @pytest.mark.parametrize('cli_args', ['--max_epochs 1 --max_steps 3 --num_nodes 1 --gpus 2'])
```
Comment (nateraw) [inline pl_examples/test_examples.py:60]: Can we remove these comments if we don't need them?

File: pl_examples/domain_templates/imagenet.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
 """
 This example is largely adapted from https://github.com/pytorch/examples/blob/master/imagenet/main.py
+
```
Comment (nateraw) [inline pl_examples/domain_templates/imagenet.py:3]: This is great info, thanks.

File: pl_examples/test_examples.py (hunk 1 above)
Comment (awaelchli) [reply pl_examples/test_examples.py:60]: Yeah I was wondering myself. They are outdated and have to be rewritten anyway if we ever get multi-node ddp testing.

**Comment by ruotianluo** [comment]
It seems the current --evaluate won't load the checkpoint from --resume_from_checkpoint, right?

I saw in the doc:
```
# (4) test with an explicit model (will use this model and not load a checkpoint)
traiener.test(model)
```

**Comment by awaelchli** [comment]
oh yes, I think you are right. Would you like to send a PR?

**Comment by ruotianluo** [comment]
I am not sure what is the most elegant solution:
One possibility would just be:
```
model.load_state_dict(torch.load(args.resume_from_checkpoint, map_location=''cpu)['state_dict'])
```
But I think it's too ugly.

**Comment by awaelchli** [comment]
we could simply pass the checkpoint path to .test
`trainer.test(model, ckpt_path="...") # default is ckpt_path="best", but that's not available`
what do you think?

**Comment by ruotianluo** [comment]
https://github.com/PyTorchLightning/pytorch-lightning/blob/8eb77cd06aff18c14242e194f6959752a9b08817/pytorch_lightning/trainer/trainer.py#L538

It won't use the ckpt_path

**Comment by awaelchli** [comment]
ok, correction, then` trainer.test(ckpt_path="...")` without model? :)

**Comment by ruotianluo** [comment]
Doesn't that assume trainer already has the model? My understanding, if you want to use this, you have to run it after trainer.fit(model) otherwise trainer doesn't know what is the model.

**Comment by awaelchli** [comment]
we could load the state at the beginning, like
```python
if args.evaluate:
    model = ImageNetLightningModel.load_from_checkpoint(path, map_location=...)
```

**Pull request merged**